src/components/JobSettingForm/JobDateFields.tsx

```diff
@@ -0,0 +1,53 @@
+import { Grid, GridProps, TextFieldProps } from '@mui/material'
+import { FormTextField } from 'components/FormComponents'
+import { DateTime } from 'luxon'
+import { TriggerType } from 'types/backend-types'
+
+
+const JobDateFields = ({ triggerType, textFieldProps, ...gridProps }: {triggerType: TriggerType, textFieldProps: TextFieldProps} & GridProps) => {
+  const gridItemProps: GridProps = {
+    item: true,
+    minWidth: '280px',
+    xs: 1
+  }
+
+  return (
+    <Grid container columns={2} {...gridProps} >
+      <Grid {...gridItemProps} >
+        <FormTextField
+          {...textFieldProps}
+          registerKey="trigger.run_date"
+          registerOptions={{
+            setValueAs: value => {
+              const dateTime = DateTime.fromISO(value)
+              return value === '' ? '' : new Date(dateTime.toHTTP()).getTime()
```

Comment:
This should not be it is only setting them to null in the other fields because they are optional fields and need to be set to `null` instead of `''` for the purpose of validation on the backend.